### PR TITLE
Disable access to the customer groups page when this feature is disabled

### LINKED ISSUE #30062: Groups could be managed even with feature disabled
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When the customers groups feature is disabled, you sill can configure add/edit some groups.

### Expected behavior

Like the features, having a warning about feature disabled.

Capture d’écran 2022-10-19 à 00 09 14

### Steps to reproduce

1. Go to performances tab
2. Disable the customers groups
3. Go to Preferences > Customer > Groups

### PrestaShop version(s) where the bug happened

8.0.0-rc.1

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by ks129**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Disables access to the customer groups page if it has been disabled on the performance page and shows a warning message.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #30062 (issue closed but still valid)
| Related PRs       |
| How to test?      | 1) Verify that everything on the customer groups page works. 2) Disable the customer groups feature. 3) Ensure that it is not possible to do anything on the customer groups page. 4) Enable the customer groups again. 5) Repeat step 1.
| Possible impacts? |

### CODE REVIEW

File: controllers/admin/AdminGroupsController.php [hunk 1]
```diff
@@ -681,4 +685,24 @@ public function displayEditLink($token, $id)
...
+     */
+    public function initContent() {
+        if (!Group::isFeatureActive()) {
+            $adminPerformanceUrl = $this->context->link->getAdminLink('AdminPerformance');
+            $url = '<a href="' . $adminPerformanceUrl . '#featuresDetachables">' . $this->trans('Performance', [], 'Admin.Global') . '</a>';
+            $this->displayWarning($this->trans('This feature has been disabled. You can activate it here: %url%.', ['%url%' => $url], 'Admin.Catalog.Notification'));
+
+            $this->context->smarty->assign([
+                'content' => $this->content,
+            ]);
```
Comment (kpodemski) [inline controllers/admin/AdminGroupsController.php:703]:
```suggestion

            ]);
```

File: controllers/admin/AdminGroupsController.php [hunk 2]
```diff
@@ -681,4 +685,24 @@ public function displayEditLink($token, $id)
 
         return $tpl->fetch();
     }
+
+    /**
+     * AdminController::initContent() override.
+     *
+     * @see AdminController::initContent()
+     */
+    public function initContent() {
```
Comment (kpodemski) [inline controllers/admin/AdminGroupsController.php]:
```suggestion
    public function initContent()
    {
```

**Review by matks — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @ks129 ,

Automated tests launched here : https://github.com/florine2623/testing_pr/actions/runs/3742560231

It looks OK ✅ 
even though the HTML tag is not well displayed https://github.com/PrestaShop/PrestaShop/issues/29940


https://user-images.githubusercontent.com/16019289/208727856-6ab4aef9-64f4-49ae-8e7d-e85c2e3fca74.mov



@matks I think you can answer this.
The id `#featuresDetachables` at the end of the link doesn't send to anywhere in particular. My question is : Should it be kept ? Or can we delete it ?

`/admin-dev/index.php/configure/advanced/performance/?_token=-5bJKyyjju94LdxJFvpfbaj6exOm0d_cq4ZyEowHWPs#featuresDetachables`

What do you think ?
Thanks!

**Comment by nicosomb** [comment]
@florine2623 I'm not @matks but if it's useless, do let's delete it 😉

**Comment by kpodemski** [comment]
Hi @ks129 

I think we can delete `#featuresDetachables`, right?

File: controllers/admin/AdminGroupsController.php [hunk 3]
```diff
@@ -681,4 +685,26 @@ public function displayEditLink($token, $id)
...
+    /**
+     * AdminController::initContent() override.
+     *
+     * @see AdminController::initContent()
+     */
+    public function initContent()
+    {
+        if (!Group::isFeatureActive()) {
+            $adminPerformanceUrl = $this->context->link->getAdminLink('AdminPerformance');
+            $url = '<a href="' . $adminPerformanceUrl . '#featuresDetachables">' . $this->trans('Performance', [], 'Admin.Global') . '</a>';
```
Comment (kpodemski) [inline controllers/admin/AdminGroupsController.php]:
```suggestion
            $url = '<a href="' . $adminPerformanceUrl . '">' . $this->trans('Performance', [], 'Admin.Global') . '</a>';
```

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
To speed things I sent a commit.

**Review by SharakPL — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @ks129 ,

I have checked this PR,

The bug is well fixed and the` #featuresDetachables `is deleted 
The automated tests are OK

[Untitled_ Jan 16, 2023 11_18 AM (2).webm](https://user-images.githubusercontent.com/111756615/212655257-88a8f071-4943-480f-aa66-7f671467ab3f.webm)

It looks OK :heavy_check_mark: 
Thank you !

**Comment by nicosomb** [comment]
Thank you @ks129 !

**Pull request merged**